### ISSUE TITLE
Connection refused (Pi3 and Firefox)

### ISSUE THREAD

**Issue by alexsmedile** [InstaPy/InstaPy#5001]
Last update of everything, not working on Pi3.

## Error message:
```
InstaPy Version: 0.6.2
 ._.  ._.  ._.  ._.  ._.  ._.  ._. 
Workspace in use: "/root/InstaPy"
Traceback (most recent call last):
  File "z_debug.py", line 5, in <module>
    from z_instabot import *
  File "/home/pi/InstaPy/z_instabot.py", line 16, in <module>
    bypass_security_challenge_using='sms')
  File "/root/.local/lib/python3.5/site-packages/instapy/instapy.py", line 321, in __init__
    self.logger,
  File "/root/.local/lib/python3.5/site-packages/instapy/browser.py", line 114, in set_selenium_local_session
    options=firefox_options,
  File "/usr/local/lib/python3.5/dist-packages/selenium/webdriver/firefox/webdriver.py", line 174, in __init__
    keep_alive=True)
  File "/usr/local/lib/python3.5/dist-packages/selenium/webdriver/remote/webdriver.py", line 157, in __init__
    self.start_session(capabilities, browser_profile)
  File "/usr/local/lib/python3.5/dist-packages/selenium/webdriver/remote/webdriver.py", line 252, in start_session
    response = self.execute(Command.NEW_SESSION, parameters)
  File "/usr/local/lib/python3.5/dist-packages/selenium/webdriver/remote/webdriver.py", line 321, in execute
    self.error_handler.check_response(response)
  File "/usr/local/lib/python3.5/dist-packages/selenium/webdriver/remote/errorhandler.py", line 242, in check_response
    raise exception_class(message, screen, stacktrace)
selenium.common.exceptions.WebDriverException: Message: connection refused
```

## My attempts to fix it:
`sudo pip3 install -U $(pip3 freeze | cut -d '=' -f 1)`

`sudo apt install tmux build-essential tk-dev libncurses5-dev libncursesw5-dev libreadline6-dev libdb5.3-dev libgdbm-dev libsqlite3-dev libssl-dev libbz2-dev libexpat1-dev liblzma-dev zlib1g-dev -y`

`sudo nano /etc/dphys-swapfile`

there I set: `CONF_SWAPSIZE=500`

Then activate swapping: `sudo systemctl start dphys-swapfile`

In addition have it start automatically: `sudo systemctl enable dphys-swapfile`

Finally, try to run the script:
`sudo xvfb-run python3 quickstart.py`

I get the same error.

Anybody knows how to fix it? Seems to run ok on Ubuntu.

**Comment by gbrehmer** [comment]
in my case update to raspbian/debian "buster" fixed all problems from switching to firefox/instapy 0.6.x.. In addition I have to manually update the geckodriver (described in the official raspberry pi how to) because instapy will download a non arm compatible version

**Comment by alexsmedile** [comment]
> 
> 
> in my case update to raspbian/debian "buster" fixed all problems from switching to firefox/instapy 0.6.x.. In addition I have to manually update the geckodriver (described in the official raspberry pi how to) because instapy will download a non arm compatible version

Thanks @gbrehmer the Rasp update fixed the problem (I started from scratch, complete reset, just to have everything as clean as possible). A few more problems started to appear tho.

I don't know why now my script runs so slow. It's been stuck here for about 10 minutes:
```
Logged in successfully!
Saving account progress...
WARNING [2019-09-09 13:41:54] [XXX]  Unable to save account progress, skipping data update Message: TypeError: window._sharedData is undefined
```
Then it started but my all Raspberry is sluggish while I run the script (which I haven't changed at all since the last update). And I seem to get this error too much frequently: `cannot get the post profile pic url`


Edit: Now it has crashed. Will try a few more times...

**Comment by alexsmedile** [comment]
After a reboot and a second attempt, this is what I get:
```
INFO [2019-09-09 13:59:10] [XXX]  Getting XXX image list...
Traceback (most recent call last):
  File "/home/pi/InstaPy/z_instabot.py", line 328, in interact_by
    randomize=True)
  File "/home/pi/.local/lib/python3.7/site-packages/instapy/instapy.py", line 5540, in interact_by_comments
    self.user_interact_media,
  File "/home/pi/.local/lib/python3.7/site-packages/instapy/instapy.py", line 2471, in interact_by_users
    media,
  File "/home/pi/.local/lib/python3.7/site-packages/instapy/like_util.py", line 450, in get_links_for_username
    browser, "profile", username, person, None, logger, logfolder
  File "/home/pi/.local/lib/python3.7/site-packages/instapy/unfollow_util.py", line 211, in get_following_status
    following_status = follow_button.text
  File "/home/pi/.local/lib/python3.7/site-packages/selenium/webdriver/remote/webelement.py", line 76, in text
    return self._execute(Command.GET_ELEMENT_TEXT)['value']
  File "/home/pi/.local/lib/python3.7/site-packages/selenium/webdriver/remote/webelement.py", line 633, in _execute
    return self._parent.execute(command, params)
  File "/home/pi/.local/lib/python3.7/site-packages/selenium/webdriver/remote/webdriver.py", line 321, in execute
    self.error_handler.check_response(response)
  File "/home/pi/.local/lib/python3.7/site-packages/selenium/webdriver/remote/errorhandler.py", line 242, in check_response
    raise exception_class(message, screen, stacktrace)
selenium.common.exceptions.StaleElementReferenceException: Message: The element reference of <button class="_5f5mN       jIbKX  _6VtSN     yZn4P   "> is stale; either the element is no longer attached to the DOM, it is not in the current frame context, or the document has been refreshed
```

Edit: Still getting this error after trying a few suggested fixes on git
```
Message: The element reference of <button class="_5f5mN       jIbKX  _6VtSN     yZn4P   "> is stale;
either the element is no longer attached to the DOM, it is not in the current frame context, or the
document has been refreshed.
```
Related to #5039

**Comment by alexsmedile** [comment]
Possible fix: #5038
